site2/docs/client-libraries-python.md

```diff
@@ -327,6 +272,198 @@ The schema definition is like this.
 }
 ```
 
+### Declare and validate schema
+
+You can send messages using `BytesSchema`, `StringSchema`, `AvroSchema`, and `JsonSchema`.
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--BytesSchema-->
+
+You can send byte data using a `BytesSchema`.
+
+**Example**
+
+```python
+producer = client.create_producer(
+                'bytes-schema-topic',
+                schema=BytesSchema())
+producer.send(b"Hello")
+
+consumer = client.subscribe(
+				'bytes-schema-topic',
+				'sub',
+				schema=BytesSchema())
+msg = consumer.receive()
+data = msg.value()
+```
+
+<!--StringSchema-->
+
+You can send string data using a `StringSchema`.
+
+**Example**
+
+```python
+producer = client.create_producer(
+                'string-schema-topic',
+                schema=StringSchema())
+producer.send("Hello")
+
+consumer = client.subscribe(
+				'string-schema-topic',
+				'sub',
+				schema=StringSchema())
+msg = consumer.receive()
+str = msg.value()
+```
+
+<!--AvroSchema-->
+
+You can declare an `AvroSchema` using one of the following methods.
+
+#### Method 1: Record
+
+You can declare an `AvroSchema` by passing a class that inherits
+from `pulsar.schema.Record` and defines the fields as
+class variables. 
+
+**Example**
+
+```python
+class Example(Record):
+    a = Integer()
+    b = Integer()
+
+producer = client.create_producer(
+                'avro-schema-topic',
+                schema=AvroSchema(Example))
+r = Example(a=1, b=2)
+producer.send(r)
+
+consumer = client.subscribe(
+				'avro-schema-topic',
+				'sub',
+				schema=AvroSchema(Example))
+msg = consumer.receive()
+e = msg.value()
+```
+
+After the producer is created, the Pulsar broker validates that the existing topic schema is "Avro" type and that the format is compatible with the schema definition of the `Example` class. If the format of the topic schema is incompatible with the schema definition, an exception occurs in the producer creation.
```

Comment:
Before creating a producer, the Pulsar broker will check the schema compatible first, if there is a schema compatible problem, an exception occurs in the producer creation.

By the way, the schema compatible check applies to all schema types, maybe we don't need to mention it in AvroSchema.